### PR TITLE
Sha checksum

### PR DESCRIPTION
**Pull request by chGoodchild**

### CODE REVIEW

File: test/test.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 import hashlib
 import time
 
+perform_slow_tests = False
```
Comment (wolfv) [inline test/test.py:10]: should this stay in? Or maybe default to true?

Locally you can run `pytest test/test.py:: test_working_download_broken_checksum` to only run a specific test.

File: test/test.py [hunk 2]
```diff
@@ -89,41 +93,69 @@ def remove_all(file):
...
-    assert Path(file["path"]).exists()
-    assert calculate_sha256("xtensor-0.24.0-hc021e02_0.tar.bz2") != "e785d6770ea5e69275c920cb1a6385bf22876e83fe5183a011d53fe705b21980"
-    assert os.path.getsize("xtensor-0.24.0-hc021e02_0.tar.bz2") != 185929
-    remove_all(file)
+    try:
+        out = subprocess.check_output([powerloader_binary,
+                                       "download",
+                                       f"{mock_server}/static/packages/{file['name']}",
+                                       "--sha",
+                                       "\"broken_checksum\""])
```
Comment (wolfv) [inline test/test.py:142]: these extra quotes are not necessary :)

File: test/test.py [hunk 3]
```diff
@@ -89,41 +93,69 @@ def remove_all(file):
...
+# Download a broken file
+def test_broken_download_good_checksum(file, powerloader_binary, mock_server):
     remove_all(file)
+
+    try:
+        out = subprocess.check_output([powerloader_binary,
+                                       "download",
+                                       f"{mock_server}/harm_checksum/static/packages/{file['name']}",
+                                       "--sha",
+                                       "\"broken_checksum\""])
```
Comment (wolfv) [inline test/test.py:157]: here as well, can remove the extra quotes.

File: test/test.py (hunk 1 above)
Comment (chGoodchild) [reply test/test.py]: Good point, I simply removed the time based test, because it might sometimes take very long...

**Pull request merged**